libevmasm/Assembly.cpp

```diff
@@ -756,3 +751,51 @@ LinkerObject const& Assembly::assemble() const
 	}
 	return ret;
 }
+
+vector<size_t> Assembly::decodeSubIds(size_t _subObjectId) const
+{
+	if (_subObjectId < m_subs.size())
+		return {_subObjectId};
+
+	auto subIdPathIt = find_if(
+			m_subPaths.begin(),
+			m_subPaths.end(),
+			[_subObjectId](auto const& subId) { return subId.second == _subObjectId; }
+	);
+
+	assertThrow(subIdPathIt != m_subPaths.end(), AssemblyException, "");
+	return subIdPathIt->first;
+}
+
+size_t Assembly::encodeSubIds(vector<size_t> const& _subIdPath)
+{
+	assertThrow(!_subIdPath.empty(), AssemblyException, "");
+	if (_subIdPath.size() == 1)
+	{
+		assertThrow(_subIdPath[0] < m_subs.size(), AssemblyException, "");
+		return _subIdPath[0];
+	}
+
+	if (m_subPaths.find(_subIdPath) == m_subPaths.end())
+	{
+		size_t objectId = numeric_limits<size_t>::max() - m_subPaths.size();
+		assertThrow(objectId >= m_subs.size(), AssemblyException, "");
+		m_subPaths[_subIdPath] = objectId;
+	}
+
+	return m_subPaths[_subIdPath];
+}
+
+Assembly const* Assembly::subAssemblyWithId(size_t _subId) const
+{
+	vector<size_t> subIds = decodeSubIds(_subId);
```

Comment:
I think it would be best to start consistently calling the vector a `subPath`:
```suggestion
	vector<size_t> subPath = decodeSubPath(_subId);
```
also
- `_subIdPath` -> `_subPath`
- `encodeSubIds()` -> `encodeSubPath()`